### DISCUSSION TITLE
Remove  IOC Dependece. Mainly in "Adonis/Lucid".

### DISCUSSION THREAD

**Discussion by Piemontez** [adonisjs/core#2777] [Ideas]
There is the possibility of abandoning the IOC?
Because of this dependency we can't use @adonis/lucid on other servers.
And Adonis/Lucid, in my opinion, has everything to overcome the sequelize and the typeorm.
But this interdependence does not allow.

**Comment by Piemontez** [comment]
Another possibility would be to just remove the dependency with adonis/core and allow a separate installation.

**Comment by avidcoder123** [comment]
What... IoC is an integral part of AdonisJS and abandoning it would cause the entire framework to burn down. Please see AdonisJS's philosophy.

**Comment by gideaoms** [comment]
Hey @Piemontez , what about this [clark-orm](https://www.npmjs.com/package/clark-orm)

**Reply by avidcoder123** [reply]
Looks really cool, now I can use this instead of Prisma (TypeORM is too unstable and buggy). Thanks for creating this!

Seems like your approach can be used for any adonis first-party package

**Comment by RomainLanz** [comment]
Hey @Piemontez! 👋 

There is no way we will drop IoC since it is part of the core of AdonisJS.
Concerning Lucid, if someone wants to create a PR that removes the hard dependency on the IoC but keeps the current DX, we are more than happy to merge it.

**Related:**

- https://github.com/adonisjs/lucid/issues/695
